batch/batch/driver/k8s_cache.py

```diff
@@ -0,0 +1,80 @@
+import time
+import asyncio
+import sortedcontainers
+
+from hailtop.utils import retry_transient_errors
+
+
+class K8sCache:
```

Comment:
This is nice. It looks like the beginning of a dict an atomic, async update operation.